src/main/java/com/johnstarich/moviematcher/models/User.java

```diff
@@ -230,13 +234,18 @@ public User addFriendToGroup(String groupName, User newMember) throws HttpExcept
 		}
 
 		Group editThisGroup = editThisGroupOptional.get();
-
+		if(editThisGroup.members.contains(newMember)) { throw new HttpException(HttpStatus.BAD_REQUEST, newMember.username + " is already in group"); }
 		groups.remove(editThisGroup);
```

Comment:
My bad Cesar, we need to make edits to this code. When I wrote this I made some major bugs ... sorry. We will look at it together.